lib/perl/Genome/Model/Tools/Analysis/Coverage/CoveragePlot.R

```diff
@@ -0,0 +1,1549 @@
+        ng <- linesGrob(x=c(at-width/3, at+width/3), y=c(d$stats[1], d$stats[1]), default.units="native", name=nn, gp=gp, vp=vp);
+        mg <- linesGrob(x=c(at-width/2, at+width/2), y=c(d$stats[3], d$stats[3]), default.units="native", name=nm, gp=gpar(lwd=1.5), vp=vp);
+        xg <- linesGrob(x=c(at-width/3, at+width/3), y=c(d$stats[5], d$stats[5]), default.units="native", name=nx, gp=gp, vp=vp);
+    }
+    
+    # TODO: creation of a child gTree object 
+    # creates an area gTree
+    g <- gTree(
+         name=name, 
+         children = gList(lg, bg, mg, ng, xg),
+         childrenvp = vp,
+         gp = gp,
+         vp = NULL,
+         cl = cl);
+    
+    # draws the outliers
+    if (outline)
+    {
+        if (horizon)
+        {
+            # adds outliers to the parent gTree
+            og <- pointsGrob(x=d$out, y=rep(at, length(d$out)), pch=21, size = unit(0.3, "char"), default.units="native", name=no, gp=gp, vp=vp);
+            g <- addGrob(g, og);
+        }
+        else
+        {
+            # adds outliers to the parent gTree
+            og <- pointsGrob(x=rep(at, length(d$out)), y=d$out, pch=21, size = unit(0.3, "char"), default.units="native", name=no, gp=gp, vp=vp);
+            g <- addGrob(g, og);
+        }
+    }
+    
+    if (draw) 
+        grid.draw(g);
+    
+    invisible(g);
+}
+
+
+cs_boxplotGrob <- function(d, at=0, width=NULL, outline=FALSE, horizon=FALSE, gp=gpar(), name=NULL, vp=NULL)
+{
+    grid.cs_boxplot(d=d, at=at, width=width, outline=outline, horizon=horizon, gp=gp, draw=FALSE, name=name, vp=vp);
+}
+
+
+
+cs_plot1 <- function(title=NULL, xlab=NULL, ylab=NULL, xscale=NULL, xfactor=0.05, yscale=NULL, yfactor=0.05, gp=gpar(), name=NULL, parent=NULL)
+{
+    # lists up the chracter identifiers for the grid components
+    if (is.null(name)) nv = "cs_plot1" else nv <- name;
+    nvr = "vruler";    # the vertical ruler
+    nhr = "hruler";    # the horizontal ruler
+
+    # the default gpar option
+    names <- names(gp);
+    if (! ("fontsize" %in% names)) gp$fontsize <- 8;
+
+    # creates a xy scatter plot region and organizes components for publication purpose
+    g <- cs_base(title=title, xlab=xlab, xmargin=3, ylab=ylab, ymargin=2, xscale=xscale, xfactor=xfactor, xvisible=T, xrot=0, xtick=0.5, xspace=-1, yscale=yscale, yfactor=yfactor, yvisible=T, yrot=0, ytick=0.5, yspace=-1, default=unit(1, "lines"), gp=gp, name=nv, parent=parent);
+    
+    # draws vertical ruler gridlines on a plot
+    g <- addGrob(g, cs_vrulerGrob(scale=g$children$xaxis$at, breaks=NULL, type="bottom", inner=T, size=0.02, lty=NULL, name=nvr, vp=g$childrenvp));
+    g <- addGrob(g, cs_hrulerGrob(scale=g$children$yaxis$at, breaks=NULL, type="left", inner=T, size=0.02, lty=NULL, name=nhr, vp=g$childrenvp));
+    
+    return(g);
+}
+
+
+
+
+# --------------------------------------
+# gets the command line arguments when run by Rscript
+args <- commandArgs(trailingOnly=TRUE);

```

Comment:
Wouldn't be a bad idea to verify that you get the expected number of arguments here.